src/components/structures/RoomSubList.js

```diff
@@ -105,8 +105,15 @@ var RoomSubList = React.createClass({
 
     applySearchFilter: function(list, filter) {
         if (filter === "") return list;
+        const lcFilter = filter.toLowerCase();
         return list.filter((room) => {
-            return room.name && room.name.toLowerCase().indexOf(filter.toLowerCase()) >= 0
+            if (room.name && room.name.toLowerCase().includes(lcFilter)) return true;
+            // only apply search filter to aliases if it looks like an alias (starts with `#`)
+            // to prevent loads of false positives with server names, e.g `matrix`
```

Comment:
This wouldn't cause false positives from server names though since you use startsWith? It would just never match anything if the filter didn't start with '#' so this is a fast-path optimisation?

Reply:
It used to be `.includes` like the above then I switched it when I added the startsWith# as we know it's start-anchored

Reply:
I can remove the comment if its too redundant

Reply:
back to you